ui/roll20_character_sheet_0.2.html

```diff
@@ -1176,6 +1177,12 @@
         <input name="attr_spellrange" class="spellrange" type="text"/>
         <input name="attr_spelltarget" class="spelltarget" type="text"/>
         <input name="attr_spellduration" class="spellduration" type="text"/>
+        <button 
+                type="action"
+                name="act_addSpellDurationToTurnOrder"
+                class="addSpellDurationToTurnOrder"
+                value="!addSpellDurationToTurnOrder @{skillname} @{spellDuration}"
```

Comment:
I believe this could work, along with functionality.
I have to experiment more on this.
I also believe this could call a function in the script block below.
Thoughts, suggestions?

Reply:
This seems like a good thing. I always weigh the added space of a new feature against the benefit. Here, the benefit clearly seems worth it.

I don't know if you can add a timer directly from a sheetworker script. You can't add a token to the turn order from there, because the sheetworker script doesn't know what token corresponds to the character. But you might be able to make a button whose value is a message that will add a non-token item to the turn order, though. In that case, you wouldn't even need to go through a sheetworker script, since button values can have expressions that access the rest of the line the button is on. Here's an example,
	<button name="roll_spellroll" class="skill-roll" type="roll"
                value="!EPRoll @{skillability}+@{skillmodifier} Tries @{skillname}: @{skillability} + @{skillmodifier}"/>

If that doesn't work, then you can have a sheetworker script trigger on the button press. As I recall, it can only modity the character. It can't even send a chat message. But it can do is set a hidden value on the sheet, which an API script watches for. That script can then make changes to the turn order.

I just uploaded the current API script. One of its functions modifies the turn order. It starts with
  on('chat:message', function(event) {

Another watches for a hidden variable to generate a chat. It starts with
  on("change:attribute", function(attribute, prev) {

Reply:
I think I'd have to add a function called `addSpellDurationToTurnOrder` that is similar to your EPRoll function.
Unsure...
I'll have to experiment a bit more, because this is all new stuff.
I'll check out your `chat` listeners too.